src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Uap.cs

```diff
@@ -15,4 +17,39 @@ internal static partial class ProcessManager
     {
         public static IntPtr GetMainWindowHandle(int processId) => IntPtr.Zero;
     }
+
+    internal static partial class NtProcessManager
+    {
+        private static ProcessModuleCollection GetModules(int processId, bool firstModuleOnly)
+        {
+            // We don't have a good way of getting all of the modules of the particular process,
+            // but we can at least get the path to the executable file for the process, and
+            // other than for debugging tools, that's the main reason consumers of Modules care about it,
+            // and why MainModule exists.
+            try
+            {
+                // Get the path to the executable
+                Process process = Process.GetProcessById(processId);
+                StringBuilder sb = new StringBuilder(1024);
+                int capacity = 1024;
+                Interop.Kernel32.QueryFullProcessImageName(process.Handle, 0, sb, ref capacity);
```

Comment:
What happens if the StringBuilder was too small? What if QueryFullProcessImageName fails for other reasons?